sklearn/tests/random_seed.py

```diff
@@ -0,0 +1,98 @@
+import pytest
+from os import environ
+from random import Random
+from datetime import datetime
+
+# Definition of the random_seed fixture. See the docstring of the fixture for
+# more information.
+
+RANDOM_SEED_RANGE = list(range(100))  # All seeds in [0, 99] should be valid.
+random_seed_var = environ.get("SKLEARN_TESTS_GLOBAL_RANDOM_SEED")
+if random_seed_var is None:
+    # This is the way.
+    random_seeds = [42]
+elif random_seed_var == "any":
+    # Pick-up one seed at random in the range of admissible random seeds.
+    #
+    # Note, to make sure that all pytest-xdist workers see the same seed, we
+    # seed the meta-random number generator with a value derived either from a
+    # CI build number variables, or from the year and the day for local builds
+    try:
+        meta_seed = int(environ["BUILD_NUMBER"])
+    except (KeyError, ValueError):
+        meta_seed = int(datetime.now().strftime("%Y%j"))
+    random_seeds = [Random(meta_seed).choice(RANDOM_SEED_RANGE)]
+elif random_seed_var == "all":
+    random_seeds = RANDOM_SEED_RANGE
+else:
+    if "-" in random_seed_var:
+        start, stop = random_seed_var.split("-")
+        random_seeds = list(range(int(start), int(stop) + 1))
+    else:
+        random_seeds = [int(random_seed_var)]
+
+    if min(random_seeds) < 0 or max(random_seeds) > 99:
+        raise ValueError(
+            "The value(s) of the environment variable "
+            "SKLEARN_TESTS_GLOBAL_RANDOM_SEED must be in the range [0, 99] "
+            f"(or 'any' or 'all'), got: {random_seed_var}"
+        )
+
+
+def pytest_report_header(config):
+    if random_seed_var == "any":
+        return [
+            "To reproduce this test run, set the following environment variable:",
+            f'    SKLEARN_TESTS_GLOBAL_RANDOM_SEED="{random_seeds[0]}"',
+            "See: https://scikit-learn.org/dev/computing/parallelism.html"
+            "#environment-variables",
+        ]
+
+
+@pytest.fixture(params=random_seeds)
+def global_random_seed(request):
+    """Fixture to ask for a random yet controllable random seed.
+
+    All tests that use this fixture accept the contract that they should
+    deterministically pass for any seed value from 0 to 99 included.
+
+    If the SKLEARN_TESTS_GLOBAL_RANDOM_SEED environment variable is set to
+    "any" (which should be the case on nightly builds on the CI), the fixture
+    will choose an arbitrary seed in the above range (based on the BUILD_NUMBER
+    or the current day) and all fixtured tests will run for that specific seed.
+    The goal is to ensure that, over time, our CI will run all tests with
+    different seeds while keeping the test duration of a single run of the full
+    test suite limited. This will check that the tests assertions of tests
```

Comment:
```suggestion
    test suite limited. This will check that the assertions of tests
```

Reply:
I removed the duplicated docstring and put a link to the online doc instead.